.github/workflows/CD.yml

```diff
@@ -27,38 +26,18 @@ jobs:
     if: github.repository == '5733d9e2be6485d52ffa08870cabdee0/sandbox'
     name: Publish JARs and Containers
     steps:
-      - name: Disk space report before modification
-        shell: bash
-        run: |
-          echo "Available storage:"
-          df -h
-      # Inspired to maximize-build-space action https://github.com/easimon/maximize-build-space
-      - name: Free disk space (remove dotnet, android and haskell)
-        shell: bash
-        run: |
-          sudo rm -rf /usr/share/dotnet
-          sudo rm -rf /usr/local/lib/android
-          sudo rm -rf /opt/ghc
-      - name: Disk space report after modification
-        shell: bash
-        run: |
-          echo "Available storage:"
-          df -h
-      - name: Setup Maven And Java Version
-        uses: s4u/setup-maven-action@v1.2.1
-        with:
-          java-version: 11
-          maven-version: '3.8.1'
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Cache Maven Packages
-        uses: actions/cache@v2
+      - name: Free disk space
+        uses: 5733d9e2be6485d52ffa08870cabdee0/sandbox/.github/actions/free-disk-space@main
+      - name: Java and Maven Setup
+        uses: 5733d9e2be6485d52ffa08870cabdee0/sandbox/.github/actions/java-maven-setup@main
         with:
-          path: ~/.m2
-          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          cache-key-prefix: ${{ runner.os }}
       - name: Build JARs
-        shell: bash
-        run: mvn --batch-mode package -Dmaven.test.skip=true -Dcheckstyle.skip
+        uses: 5733d9e2be6485d52ffa08870cabdee0/sandbox/.github/actions/maven@main
+        with:
+          maven-command: mvn --batch-mode package -Dmaven.test.skip=true -Dcheckstyle.skip
```

Comment:
What about move `mvn` inside the action? Currently it is possible to run any bash command :)

Reply:
Fine for me.